reactor-core/src/main/java/reactor/core/publisher/FluxBufferPredicate.java

```diff
@@ -176,8 +178,13 @@ public void request(long n) {
 
 		@Override
 		public void cancel() {
+			C b;
+			synchronized (this) {
```

Comment:
why not getAndSet?

Reply:
I should probably eliminate the volatile above. overall for this issue `getAndSet` doesn't cut it because the `onNext` could still capture the `buffer` just in time and perform the `add` despite cancel swapping out the buffer.